source/components/triggering-changes-with-actions.md

```diff
@@ -156,3 +189,192 @@ a function instead.
 Actions in components allow you to
 decouple an event happening from how it's handled, leading to modular,
 more reusable components.
+
+## Handling Action Completion
+
+Often actions perform asynchronous tasks, such as making an ajax request to a server.
+Since actions are functions that can be passed in by a parent component, they are able to return values when called.
+The most common scenario is for an action to return a promise so that the component can handle the action's completion.
+
+In our user `button-with-confirmation` component we want to leave the confirmation modal open until we know that the
+operation has completed successfully.
+This is accomplished by expecting a promise to be returned from `onConfirm`.
+Upon resolution of the promise, we set a property used to indicate the visibility of the confirmation modal.
+
+```app/components/button-with-confirmation.js
+export default Ember.Component.extend({
+  actions: {
+    launchConfirmDialog() {
+      this.set('confirmShown', true);
+    },
+
+    submitConfirm() {
+      //call onConfirm with the value of the input field as an argument
+      const promise = this.get('onConfirm')();
+      promise.then(() => {
+        this.set('confirmShown', false);
+      });
+    },
+
+    cancelConfirm() {
+      this.set('confirmShown', false);
+    }
+  }
+});
+```
+
+## Passing Arguments
+
+Actions passed to a component may be invoked with arguments.
+In the `send-message` component, we want to provide a message type to send to the action.
```

Comment:
@toddjordan I've just read through this carefully (and am learning new things as I read :smile: ).  I think I'm following what is going on, but as someone being exposed to this for the first time, I think it'd be helpful to have some additional context as to why I would want to do this.  Could you put in a bit more detail in here somewhere about the reason you might want to use this?  Otherwise I'm left feeling like it's something I can do, but not sure why I would.